app/controllers/course/survey/responses_controller.rb

```diff
@@ -1,27 +1,14 @@
 # frozen_string_literal: true
 class Course::Survey::ResponsesController < Course::Survey::SurveysController
   load_and_authorize_resource :response, through: :survey, class: Course::Survey::Response.name,
-                                         except: :respond
+                                         only: [:show, :update]
 
-  def respond
+  def find_or_create
```

Comment:
I believe it's better to split this into `create` and `show`, the backend should be general enough to work with any kind of frontend. 

I think in this case, we can response with a created status and the id of the created survey if errored ?  and the frontend should then issue a get request in the error recovery to get the created response.